cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeJobCleaner.java

```diff
@@ -82,6 +88,9 @@ public void run() {
           }
         }
         continuationToken = jobs.getMetadata().getContinue();
+      } catch (IOException e) {
+        LOG.warn("Failed to cleanup jobs. Error creating batch api client.", e);
+        break;
```

Comment:
Should we try on this error as well? There could be temporary disconnection to the API server?

Reply:
hmm, I thought this was only looking at kubeconfig files cached locally, but it does reach out to the server as a last resort. We can retry on this error.